### PR TITLE
[Flex] Added `Flex` Component

### PR DESCRIPTION
**Pull request by rwu823**
image

### CODE REVIEW

File: src/client/views/flex/index.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        <h2>Auto</h2>
+        <Flex>
+          <Flex.Col size="auto"><p class={[s.p]}>size="auto"</p></Flex.Col>
+          <Flex.Col size="auto"><p class={[s.p]}>size="auto"</p></Flex.Col>
+          <Flex.Col size="auto"><p class={[s.p]}>size="auto"</p></Flex.Col>
+        </Flex>
+
+        <h2>12 Grids</h2>
+        <div class={s.boxs}>
+          { [...Array(6)].map((v, i) => {
```
Comment (rwu823) [inline src/client/views/flex/index.js:22]: 神 code 乍現 QQ

Comment (chuanxd) [reply src/client/views/flex/index.js:22]: XD

File: src/components/flex/flex.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    &:last-child {
+      margin-right: 0!important;
+    }
+  }
+}
+
+.flexMid {
+  display: flex;
+  align-items: center;
+  justify-content: center;
```
Comment (chuanxd) [inline src/components/flex/flex.scss:13]: 這真的很方便

File: src/components/flex/flex.scss [hunk 3]
```diff
@@ -0,0 +1,23 @@
+.flex {
+  display: flex;
+  > div {
+    &:last-child {
+      margin-right: 0!important;
```
Comment (rwu823) [inline src/components/flex/flex.scss]: 這邊 `!important` 是必備，先講一下 QQ

File: src/components/flex/flex.scss (hunk 2 above)
Comment (CApopsicle) [reply src/components/flex/flex.scss:13]: 對呀這我也很常用

File: src/client/views/flex/flex.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
+.p {
+  margin: 0;
+  padding: 0;
+  background: lighten(#590AAA, 50%);
```
Comment (chuanxd) [inline src/client/views/flex/flex.scss]: lowercase `#590aaa` 我們統一用小寫

File: src/components/flex/flex.scss (hunk 3 above)
Comment (chuanxd) [reply src/components/flex/flex.scss]: ok
space QQ
`margin-right: 0 !important;`

File: src/client/views/flex/doc.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  return (
+    <Flex grids={24}>
+
+    </Flex>
+  )
+}
+```
+
+## Margin
+Set the margin of each column, default is `20`.
```
Comment (chuanxd) [inline src/client/views/flex/doc.md:88]: 說明加上單位會不會比較清楚呢？ `px`

File: src/components/flex/index.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  computed: {
+    style() {
+      const { $parent, size, offset } = this
+
+      const styleObj = {
+        marginRight: `${$parent.margin}px`,
+      }
+
+      if (size === 'auto') {
+        styleObj.flex = 1
```
Comment (chuanxd) [inline src/components/flex/index.js:67]: 這邊直接做等分嗎？
如果像是測試裡面的這段
```jsx
<Flex>
  <Flex.Col size="2" />
  <Flex.Col size="auto" />
</Flex>
```

render 後 column 會都為 `6` ?

**Review by chuanxd — CHANGES_REQUESTED** [review]

File: src/components/flex/index.js (hunk 6 above)
Comment (rwu823) [reply src/components/flex/index.js:67]: If in 12 grids set `2` and `auto` it should be rendered as `2` and `10`

Comment (rwu823) [reply src/components/flex/index.js:67]: `auto` is mean automate flexible

Comment (rwu823) [reply src/components/flex/index.js:67]: `auto * 2` it'll be `6` `6`     

`auto * 3` it'll be `4` `4` `4`

File: src/client/views/flex/flex.scss (hunk 4 above)
Comment (rwu823) [reply src/client/views/flex/flex.scss]: why no exception in stylelint QQ

File: src/components/flex/index.js (hunk 6 above)
Comment (chuanxd) [reply src/components/flex/index.js:67]: got it.

File: src/client/views/flex/flex.scss (hunk 4 above)
Comment (chuanxd) [reply src/client/views/flex/flex.scss]: https://github.com/cepave-f2e/vue-owl-ui/blob/dev/.stylelintrc.js#L5

I had set.

Comment (chuanxd) [reply src/client/views/flex/flex.scss]: @rwu823 CI lint missing ?

**Comment by rwu823** [comment]
All suggestions are fixed.

**Comment by chuanxd** [comment]
自動被忽略了QQ

**Pull request merged**